newmodel.py

```diff
@@ -482,39 +482,44 @@ def clear_input_form():
 
 from ipywidgets import FileUpload
 import os
-upload = FileUpload()
+#upload = FileUpload(multiple=True)
 class observe_file_upload:
     def __init__(self):
         self.metadata = None
-        self.name = None
+        self.name = []
     def __call__(self,change):
         n = change["name"]
         v = change["new"]
         if n == "metadata":
             # Not sure why this sometimes
             # comes in as a list and sometimes
             # not.
-            if type(v) == list:
-                self.metadata = v[0]
-            else:
-                self.metadata = v
-            self.name = self.metadata["name"]
+            for i in range(len(v)):
+                if type(v) == list:
```

Comment:
The loop implies this is a list... what else are you expecting here? A tuple? Please add some comments.